scraper.py

```diff
@@ -189,3 +196,139 @@ def get_sorted_list(self, sort_by="rank", order="descending", limit=None, item_t
+                        ]
+                    except ValueError:
+                        print("Failed to Convert")  # If the value cannot be converted to float, ignore the filter
+                else:
+                    # For string or other non-numeric fields, perform substring match
+                    filtered_items = [
+                        item for item in filtered_items if value.lower() in str(item.get(field, "")).lower()
+                    ]
+
+        # Limit the number of results if a limit is specified
+        if limit is not None:
+            filtered_items = filtered_items[:limit]
+
+        # If sort_by is provided, sort the filtered items
+        if sort_by and sort_by in self.items[0]:
+            print("Sorting")
+            filtered_items = sorted(
+                filtered_items, 
+                key=lambda item: self.safe_sort_key(item, sort_by),
+                reverse=(order == "descending")
+            )        
+
+        return filtered_items 
+    
+    def safe_sort_key(self, item, sort_by):
+        """
+        Safely handles sorting for both numeric and string fields.
+        """
+        value = item.get(sort_by)
+        if isinstance(value, (int, float)):
+            return value
+        
+        try:
+            return float(value)
+        except (ValueError, TypeError):
+            pass
+
+
+        if isinstance(value, str):
+            if value.strip().lower() in ["na", "none","n/a", ""]:
+                return(float('inf'))
+            return value.lower()
+        else:
+            return float('inf')  # If the value is not numeric or string, place it at the end.    
+
+        
+
+
+    def safe_compare(self, field_value, filter_value):
+        #Compares field values to filter values
+
+        if isinstance(field_value, (int, float)):
+            return field_value == filter_value
+        
+        elif isinstance(field_value, str):
+            return filter_value.lower() in field_value.lower()
+        
+        return False
+
+    def export_to_csv(self, data, filename="output.csv", fields=None):
+        """
+        Exports a list of dictionaries to a CSV file, with the option to limit columns.
+        
+        :param data: List of dictionaries containing the data to export.
+        :param filename: The name of the CSV file to save.
+        :param fields: List of field names to include in the CSV. If None, all fields are included.
+        """
+        if not data:
+            print("No data to export.")
+            return
+        
+        # Use specified fields or default to all keys from the first item
+        if fields:
+            # Ensure fields exist in the data
+            headers = [field for field in fields if field in data[0]]
+        else:
+            headers = data[0].keys()
+
+        try:
+            with open(filename, mode="w", newline="", encoding="utf-8") as file:
```

Comment:
Shouldn't this export the CSV to a string? It should be up to the application to wrtite the file.